framework/Models/PostProcessors/PostProcessorInterface.py

```diff
@@ -0,0 +1,134 @@
+    self.outputMultipleRealizations = False
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the common parts of the model parameter input.
+      @ In, paramInput, InputData.ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    super()._handleInput(paramInput)
+
+  def initialize(self, runInfo, inputs, initDict=None):
+    """
+      Method to initialize the PostProcessor
+      @ In, runInfo, dict, it is the run info from the jobHandler
+      @ In, inputs, list, it is a list containing whatever is passed with an input role in the step
+      @ In, initDict, dict, optional, dictionary of all objects available in the step is using this model
+      @ Out, None
+    """
+    super().initialize()
+    self.inputs = inputs
+    if 'stepName' in runInfo:
+      self.workingDir = os.path.join(runInfo['WorkingDir'],runInfo['stepName']) #generate current working dir
+    else:
+      self.workingDir = runInfo['WorkingDir']
+
+  @abc.abstractmethod
+  def run(self, input):
+    """
+      This method executes the postprocessor action.
+      @ In,  input, object, object containing the data to process.
+      Should avoid to use (inputToInternal output), and passing xarray directly/dataset
+      Possible inputs include: dict, xarray.Dataset, pd.DataFrame
+      @ Out, dict, xarray.Dataset, pd.DataFrame --> I think we can avoid collectoutput in the plugin pp
+    """
+
+  def collectOutput(self,finishedJob, output, options=None):
+    """
+      Method that collects the outputs from the "run" method of the PostProcessor
+      @ In, finishedJob, InternalRunner object, instance of the run just finished
+      @ In, output, "DataObjects" object, output where the results of the calculation needs to be stored
+      @ In, options, dict, optional, not used in PostProcessor.
+        dictionary of options that can be passed in when the collect of the output is performed by another model (e.g. EnsembleModel)
+      @ Out, None
+    """
+    if output.type not in self.validDataType:
+      self.raiseAnError(IOError, 'Output type', str(output.type), 'is not allowed!')
+    evaluation = finishedJob.getEvaluation()
+    if isinstance(evaluation, Runners.Error):
+      self.raiseAnError(RuntimeError, "No available output to collect (run possibly not finished yet)")
+    outputRealization = evaluation[1]
+
+    if output.type in ['PointSet','HistorySet']:
+      if self.outputDataset:
+        self.raiseAnError(IOError, "DataSet output is required, but the provided type of DataObject is", output.type)
+      self.raiseADebug('Dumping output in data object named ' + output.name)
+      if self.outputMultipleRealizations:
+        if 'dims' in outputRealization:
+          dims = outputRealization['dims']
+        else:
+          dims = {}
+        print(outputRealization.keys())
+        output.load(outputRealization['data'], style='dict', dims=dims)
```

Comment:
the load should be modified to also accept a single realizaiton.

PS. I think that the PP must return a dataset. dictionaries are slow

Reply:
Let's have a discussion for this.

Reply:
we can, but the "gate" (if statement) should be included in the load method